src/Microsoft.Management.Configuration/ConfigurationProcessor.cpp

```diff
@@ -958,6 +958,49 @@ namespace winrt::Microsoft::Management::Configuration::implementation
         return *result;
     }
 
+    Windows::Foundation::Collections::IVector<IConfigurationUnitProcessorDetails> ConfigurationProcessor::FindUnitProcessors(const FindUnitProcessorsOptions& findOptions)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+        return FindUnitProcessorsImpl(findOptions);
+    }
+
+    Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVector<IConfigurationUnitProcessorDetails>> ConfigurationProcessor::FindUnitProcessorsAsync(const FindUnitProcessorsOptions& findOptions)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+
+        auto strong_this{ get_strong() };
+        FindUnitProcessorsOptions localOptions = findOptions;
+
+        co_await winrt::resume_background();
+
+        co_return FindUnitProcessorsImpl(localOptions, { co_await winrt::get_cancellation_token() });
+    }
+
+    Windows::Foundation::Collections::IVector<IConfigurationUnitProcessorDetails> ConfigurationProcessor::FindUnitProcessorsImpl(
+        const FindUnitProcessorsOptions& findOptions,
+        AppInstaller::WinRT::AsyncCancellation cancellation)
+    {
+        auto threadGlobals = m_threadGlobals.SetForCurrentThread();
+
+        IConfigurationSetProcessor setProcessor = m_factory.CreateSetProcessor(nullptr);
+        auto result = winrt::single_threaded_vector<IConfigurationUnitProcessorDetails>();
+
+        cancellation.ThrowIfCancelled();
+
+        IFindUnitProcessorsSetProcessor findUnitProcessorsSetProcessor;
+
+        if (setProcessor.try_as<IFindUnitProcessorsSetProcessor>(findUnitProcessorsSetProcessor))
+        {
+            result = findUnitProcessorsSetProcessor.FindUnitProcessors(findOptions);
+        }
+        else
+        {
+            AICLI_LOG(Config, Warning, << "Set Processor does not support FindUnitProcessors operation");

```

Comment:
I feel like this should be an error to indicate that the operation could not be performed.  We already do similar with the `GetAll*` functions.

Reply:
I meant that the API should produce an error.